### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by asiripanich**
- [x] #1 
- [x] #2 
- [x] #4
- [x] #5
- [x] #14 
- [x] #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,26 @@ To learn more about Dash, please visit [documentation](https://plot.ly/dash).
...
+
+
+### STUDY_NAME
+
+The `STUDY_NAME` environment variable is used to specify the name of the study or program that is being run inside the
+Docker container. This variable is typically used by the application running inside the container to differentiate
+between different studies or programs.
+
+Note that the `STUDY_NAME` variable can be set to any string value, and should be set to a unique value for each
+separate study or program.
```
Comment (shankari) [inline README.md:91]: I don't want to use the STUDY_NAME variable. This is a webapp; it can automatically infer the name from the URL. Having two sources of truth is asking for corner case issues.
Please see https://github.com/e-mission/op-admin-dashboard/pull/7#discussion_r1142882608

File: app_sidebar_collapsible.py [hunk 2]
```diff
@@ -200,16 +200,18 @@ def update_store_trips(start_date, end_date):
     Input('url', 'search'),
 )
 def display_page(search):
-    try:
-        if auth_type == 'cognito':
+    if auth_type == 'cognito':
+        try:
```
Comment (shankari) [inline app_sidebar_collapsible.py:204]: why did you need to change the scope of the `if` block?

File: .gitignore [hunk 3]
```diff
@@ -1,10 +1,5 @@
 __pycache__
-.DS_Store
```
Comment (shankari) [inline .gitignore]: you can keep it in the `.gitignore` (it is a file autogenerated by OSX), it just shouldn't be in the repo

File: .dockerignore [hunk 4]
```diff
@@ -1 +1 @@
-**/*/Dockerfile
```
Comment (shankari) [inline .dockerignore]: Is this entire commit (https://github.com/e-mission/op-admin-dashboard/pull/7/commits/185e5e94c44217b1b83ed6ff6f0cc80fff163721#diff-2f754321d62f08ba8392b9b168b83e24ea2852bb5d815d63e767f6c3d23c6ac5L1) just adding a final line to each file?

File: docker-compose-prod.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# docker-compose-prod.yml
+version: "3"
+services:
+  dashboard:
+    build:
+      context: .
+      dockerfile: docker/Dockerfile:0.0.1
+    image: op-admin-dash-prod/opdash
```
Comment (shankari) [inline docker-compose-prod.yml:8]: Ditto

File: docker-compose-dev.yml [hunk 6]
```diff
@@ -5,7 +5,7 @@ services:
     build:
       context: .
       dockerfile: docker/Dockerfile
-    image: openpath/opdash:0.0.1
+    image: op-admin-dash-dev/opdash:0.0.1
```
Comment (shankari) [inline docker-compose-dev.yml:8]: There is no such organization either.
Put it in `e-mission` or in `shankari`

File: docker-compose-prod.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    ports:
+      - "8050:8050"
+    environment:
+      DASH_DEBUG_MODE: "False"
+      DASH_SERVER_PORT: 8050
+      DB_HOST: db
+      WEB_SERVER_HOST: 0.0.0.0
+      SERVER_BRANCH: master
+      CONFIG_PATH: "https://raw.githubusercontent.com/AlirezaRa94/nrel-openpath-deploy-configs/main/configs/"
+      STUDY_NAME: "admin-dashboard-permissions"
```
Comment (shankari) [inline docker-compose-prod.yml:18]: please remove `STUDY_NAME`
https://github.com/e-mission/op-admin-dashboard/pull/7#discussion_r1142882608

File: utils/db_utils.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    projection = {
+        '_id': 0,
+        'user_id': 1,
+        'trip_start_time_str': '$data.start_fmt_time',
+        'trip_end_time_str': '$data.end_fmt_time',
+        'timezone': '$data.start_local_dt.timezone',
+        'start_coordinates': '$data.start_loc.coordinates',
+        'end_coordinates': '$data.end_loc.coordinates',
+        'travel_modes': '$data.user_input.trip_user_input.data.jsonDocResponse.data.travel_mode',
```
[the remark this answers is not in the document]
Comment (shankari) [reply utils/db_utils.py:58]: so I think this is the list of columns to you want to extract from the `confirmed_trips` "table" (collection in mongodb).
Note that you should really be using the methods from the timeseries here instead of running your own projection code. If the methods you need don't exist, ask for them.

Regardless, either as a projection on the DB itself, or after you have read it into a pandas dataframe, you will need to select the columns that you need to display.

Some of these columns can and should be standard, such as the distance, duration, and so on.
Some of the other columns should be read from the dynamic config. 
So the config should have an entry for "additional_columns" and that will represent the strings mapping into the document for this study or program.

So for UNSW studies, this will be `$data.user_input.trip_user_input.data.jsonDocResponse.data.travel_mode`
For MULTILABEL studies, this will be `$data.user_input.mode_confirm.label`
For some other study, where they don't even care about the mode but only about the amount of range anxiety
`$data.user_input.trip_user_input.jsonDocResponse.data.range_anxiety`

This could look like

```
additionalColumns: {
   {label: 'range_anxiety', 'path': $data.user_input.trip_user_input.jsonDocResponse.data.range_anxiety
}
```

File: docker-compose-dash-app.yml [hunk 9]
```diff
@@ -1,13 +1,47 @@
...
+    deploy:
+      replicas: 1
+      restart_policy:
+        condition: on-failure
+    volumes:
+      - mongo-data:/data/db
+    networks:
+       - emission
+    ports:
+      - "27017:27017"
```
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply docker-compose-dash-app.yml]: To be able to connect to the database from my local machine. I've removed it in the production version of the docker-compose. [e4c533e](https://github.com/e-mission/op-admin-dashboard/pull/7/commits/e4c533ec6fe9fef7c5c6a1106bb3eac71c864713)

File: .dockerignore (hunk 4 above)
Comment (AlirezaRa94) [reply .dockerignore]: I thought there were other changes in this commit, but I didn't add them by mistake, and they are in the other commit.

File: app_sidebar_collapsible.py (hunk 2 above)
Comment (AlirezaRa94) [reply app_sidebar_collapsible.py:204]: In the authentication process, if something is wrong (like the AWS Cognito credentials), we shouldn't show the home page, and that's why I've used `try` here.

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,10 @@
+__pycache__
+.DS_Store
+.idea/
+emission/
+conf/
+bin/
```
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply .gitignore]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/f0a5fe7f9ba6dd03eeb0d38df69e5cc61de8f019

File: docker-compose-dash-app.yml [hunk 11]
```diff
@@ -1,13 +1,47 @@
...
+      CONFIG_PATH: "https://raw.githubusercontent.com/AlirezaRa94/nrel-openpath-deploy-configs/main/configs/"
+      STUDY_NAME: "admin-dashboard-permissions"
+      AUTH_TYPE: "basic" # the other option is cognito
+    networks:
+       - emission
+    volumes:
+      - ./pages:/usr/src/app/pages
+      - ./utils:/usr/src/app/utils
+      - ./app.py:/usr/src/app/app.py
+      - ./app_sidebar_collapsible.py:/usr/src/app/app_sidebar_collapsible.py
```
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply docker-compose-dash-app.yml]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/e4c533ec6fe9fef7c5c6a1106bb3eac71c864713

File: docker-compose-prod.yml [hunk 12]
```diff
@@ -0,0 +1,37 @@
+# docker-compose-prod.yml
+version: "3"
+services:
+  dashboard:
+    build:
+      context: .
+      dockerfile: docker/Dockerfile:0.0.1
```
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply docker-compose-prod.yml]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/62307f87190e87e9ee7e390b41681120d79f613f

File: docker-compose-dev.yml (hunk 6 above)
Comment (AlirezaRa94) [reply docker-compose-dev.yml]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/62307f87190e87e9ee7e390b41681120d79f613f

File: docker-compose-prod.yml (hunk 5 above)
Comment (AlirezaRa94) [reply docker-compose-prod.yml]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/62307f87190e87e9ee7e390b41681120d79f613f

File: .gitignore (hunk 3 above)
Comment (AlirezaRa94) [reply .gitignore]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/62307f87190e87e9ee7e390b41681120d79f613f

File: docker-compose-dash-app.yml [hunk 13]
```diff
@@ -1,13 +1,47 @@
...
 version: "3"
 services:
-  dash-app:
+  dashboard:
     build:
       context: .
-      dockerfile: ./Dockerfile
-    image: dav/dash-app:0.0.1
+      dockerfile: docker/Dockerfile
+    image: openpath/opdash:0.0.1
```
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply docker-compose-dash-app.yml]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/62307f87190e87e9ee7e390b41681120d79f613f

File: docker-compose-prod.yml (hunk 12 above)
[the remark this answers is not in the document]
Comment (shankari) [reply docker-compose-prod.yml]: Can you indicate the "Testing done" in your commits? Would be useful to see if this actually worked, for example

File: app_sidebar_collapsible.py (hunk 2 above)
Comment (shankari) [reply app_sidebar_collapsible.py:204]: And you have now moved both `get_cognito_login_page` to the exception handling? Why?

Comment (AlirezaRa94) [reply app_sidebar_collapsible.py:204]: > Right, but before this change, you had
> 
> ```
> try:
>     if
> ```
> 
> now you have
> 
> ```
> if
>    try:
> ```
> 
> Both of them seem to have the same functionality, so just trying to understand the reason for the change.

These changes are to increase the code readability. Other parts also didn't need to be inside the exception handling.

Comment (AlirezaRa94) [reply app_sidebar_collapsible.py:204]: > And you have now moved both `get_cognito_login_page` to the exception handling? Why?

The second one is outside of exception handling.

File: utils/db_utils.py [hunk 14]
```diff
@@ -0,0 +1,72 @@
+from datetime import datetime, timezone
+
+import pandas as pd
+
+import emission.core.get_database as edb
```
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply utils/db_utils.py:5]: Please provide the link to the Timeseries_sample notebook that you've mentioned.

[the remark this answers is not in the document]
Comment (shankari) [reply utils/db_utils.py:5]: There are examples all over the e-mission codebase as well. Timeseries_sample (from the e-mission-server repo) has a high level overview, but you can just look at the classes in `emission/storage/timeseries` and `emission/storage/decorations`

File: utils/db_utils.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply utils/db_utils.py]: solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/8a3c37da21c10652f593d1a6d2be8dacb1aa47b2

**Comment by shankari** [comment]
@AlirezaRa94 any updates on when these changes will done?

**Comment by AlirezaRa94** [comment]
> @AlirezaRa94 any updates on when these changes will done?

I'm working on them and will push the code when finished.

**Comment by shankari** [comment]
> I'm working on them and will push the code when finished.

Do you have an ETA or timeline?
In particular, once the `STUDY_NAME` changes are done, we can at least try to deploy on staging and standardize our deployment procedure.

File: utils/db_utils.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (AlirezaRa94) [reply utils/db_utils.py:5]: I couldn't find equivalent functions. If there are these functions, please give me the full path.

**Comment by shankari** [comment]
Also, what (if any), are the changes in 
https://github.com/e-mission/op-admin-dashboard/pull/7/commits/8a3c37da21c10652f593d1a6d2be8dacb1aa47b2#diff-9f23c8b13c9e75b67a44235089ab45e194bb00a6339db3295e9ff6a0646f2cc7

Is it just whitespace changes?
It would be good to avoid mixing whitespace changes with intentional code changes to make the changes easier to review.

**Comment by AlirezaRa94** [comment]
> > I'm working on them and will push the code when finished.
> 
> Do you have an ETA or timeline? In particular, once the `STUDY_NAME` changes are done, we can at least try to deploy on staging and standardize our deployment procedure.

To change the `STUDY_NAME`, I have to make changes in most parts of the code, and I should test everything again. It takes some time, but I will try to finish it soon.

**Comment by AlirezaRa94** [comment]
> Also, what (if any), are the changes in [8a3c37d#diff-9f23c8b13c9e75b67a44235089ab45e194bb00a6339db3295e9ff6a0646f2cc7](https://github.com/e-mission/op-admin-dashboard/commit/8a3c37da21c10652f593d1a6d2be8dacb1aa47b2#diff-9f23c8b13c9e75b67a44235089ab45e194bb00a6339db3295e9ff6a0646f2cc7)
> 
> Is it just whitespace changes? It would be good to avoid mixing whitespace changes with intentional code changes to make the changes easier to review.

Yes, It's just white spaces.
Sure.

**Comment by shankari** [comment]
> To change the `STUDY_NAME`, I have to make changes in most parts of the code, and I should test everything again. It takes some time, but I will try to finish it soon.

Why? you only read the environment variable in one location, so you should be able to only read the study config from the location in one place.

**Comment by AlirezaRa94** [comment]
> > To change the `STUDY_NAME`, I have to make changes in most parts of the code, and I should test everything again. It takes some time, but I will try to finish it soon.
> 
> Why? you only read the environment variable in one location (`permissions.py`), so you should be able to only read the study config from the location in one place.
> 
> ```
> STUDY_NAME = os.getenv('STUDY_NAME')
> ```

It was an environment variable, but now I have to use a callback to get the `STUDY_NAME` from URL, and I don't have access to it in the `permissions.py` file.

**Comment by shankari** [comment]
> It was an environment variable, but now I have to use a callback to get the STUDY_NAME from URL, and I don't have access to it in the permissions.py file.

There must be a way to transfer the data from the callback to `permissions.py`.
- You can make a call to `permissions.py` to set the study name
- you can set the environment variable in the callback
- ...

**Comment by AlirezaRa94** [comment]
> > It was an environment variable, but now I have to use a callback to get the STUDY_NAME from URL, and I don't have access to it in the permissions.py file.
> 
> There must be a way to transfer the data from the callback to `permissions.py`.
> 
> * You can make a call to `permissions.py` to set the study name
> * you can set the environment variable in the callback
> * ...

Thank you, If you are OK with these methods, I will make changes using them.

**Comment by shankari** [comment]
> Thank you, If you are OK with these methods, I will make changes using them.

The ideal way is to register the callback from `permissions.py` so that you don't have unnecessary dependencies
But **if that is not possible**, then the methods above are OK.
And are strictly better than sprinkling the location callback across multiple locations in the code, if that is what you wanted to do.

For completeness, if registering the callback from `permissions.py` is not possible, please document what you tried and why it didn't work.

**Comment by AlirezaRa94** [comment]
> > Thank you, If you are OK with these methods, I will make changes using them.
> 
> The ideal way is to register the callback from `permissions.py` so that you don't have unnecessary dependencies But **if that is not possible**, then the methods above are OK. And are strictly better than sprinkling the location callback across multiple locations in the code, if that is what you wanted to do.
> 
> For completeness, if registering the callback from `permissions.py` is not possible, please document what you tried and why it didn't work.

We can create a callback inside `permissions.py`. However, the callback won't automatically run when the module is loaded because it is outside the application script. So we'll need to trigger it somehow inside the application script, which is not different from setting it outside the module.

**Comment by shankari** [comment]
@AlirezaRa94 any updates? it's been a week since the last update and we need to deploy this...

**Comment by AlirezaRa94** [comment]
@shankari I'm busy these days, but I'll try to find some free time to work on it this week.

**Comment by shankari** [comment]
@AlirezaRa94 the goal was to have it fully done and ready to deploy by end of March. It is now mid-Apr, and we need to be showing off demos...

**Comment by AlirezaRa94** [comment]
> @AlirezaRa94 the goal was to have it fully done and ready to deploy by end of March. It is now mid-Apr, and we need to be showing off demos...

The goal was to have the time-use feature and the dashboard done by the end of February. The dashboard with authentication and dynamic config was ready at that time. At that time, my task was working full-time on the dashboard. But now I have other duties and must find some free time to work on the dashboard. 
Right now, I'm working on your comments which I got on 21 March. However, I've addressed most of your comments. 
Removing the `STUDY_NAME` from the docker-compose file is not as simple as you think, which I will explain why in another comment.

**Comment by AlirezaRa94** [comment]
@shankari 
Now that we have `STUDY_NAME` in the docker-compose, we have the dynamic config before loading the dashboard layouts. But if we want to infer it from the URL, we need to get it in a request, and this happens after loading the dashboard, which will break everything.
To solve this problem, we can add a welcome page and set the dynamic config at this time, and then we can load the dashboard. 
I'm also open to hearing your ideas on this.

File: utils/db_utils.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (shankari) [reply utils/db_utils.py:5]: @AlirezaRa94 couldn't find equivalent functions for which calls?
If you list out the calls, I can give you a mapping.

This is particularly important since the current methods in here are causing errors while being used with DocumentDB, a MongoDB-compatible-ish hosted solution that is used at NREL.

We test the timeseries functionality against DocumentDB, but arbitrary queries may or may not work.
We are not able to deploy the dashboard on staging because of this issue.

```
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/dash/_callback.py", line 447, in add_context
output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
File "app_sidebar_collapsible.py", line 190, in update_store_trips
df = query_confirmed_trips(start_date_obj, end_date_obj)
File "/usr/src/app/utils/db_utils.py", line 66, in query_confirmed_trips
df = pd.json_normalize(list(query_result))
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1207, in next
if len(self.__data) or self._refresh():
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1124, in _refresh
self.__send_message(q)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1001, in __send_message
address=self.__address)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1372, in _run_operation_with_response
exhaust=exhaust)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1471, in _retryable_read
return func(session, server, sock_info, slave_ok)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1366, in _cmd
unpack_res)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/server.py", line 137, in run_operation_with_response
first, sock_info.max_wire_version)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/helpers.py", line 168, in _check_command_response
max_wire_version)
pymongo.errors.OperationFailure: Internal server error, full error: {'ok': 0.0, 'operationTime': Timestamp(1681425547, 1), 'code': 42, 'errmsg': 'Internal server error'}
```

**Comment by shankari** [comment]
@AlirezaRa94 

> Right now, I'm working on your comments which I got on 21 March.

That's because the dynamic config was only done on 20th March (https://github.com/e-mission/op-admin-dashboard/pull/7/commits/dbae8f9992bdee927be54aefcbf2084817f0ae51)

> However, I've addressed most of your comments.

You do need to make the changes to use the timeseries functions instead of a `db_util` - we are getting an error with the queries you are using 

```
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/dash/_callback.py", line 447, in add_context
output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
File "app_sidebar_collapsible.py", line 190, in update_store_trips
df = query_confirmed_trips(start_date_obj, end_date_obj)
File "/usr/src/app/utils/db_utils.py", line 66, in query_confirmed_trips
df = pd.json_normalize(list(query_result))
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1207, in next
if len(self.__data) or self._refresh():
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1124, in _refresh
self.__send_message(q)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1001, in __send_message
address=self.__address)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1372, in _run_operation_with_response
exhaust=exhaust)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1471, in _retryable_read
return func(session, server, sock_info, slave_ok)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1366, in _cmd
unpack_res)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/server.py", line 137, in run_operation_with_response
first, sock_info.max_wire_version)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/helpers.py", line 168, in _check_command_response
max_wire_version)
pymongo.errors.OperationFailure: Internal server error, full error: {'ok': 0.0, 'operationTime': Timestamp(1681425547, 1), 'code': 42, 'errmsg': 'Internal server error'}
```

Note that we internally use DocumentDB a managed solution that is supposed to be compatible with MongoDB but sometimes has minor differences. We keep (and will continue to keep) the timeseries code consistent with both MongoDB and DocumentDB, but having people write arbitrary code in other modules makes it hard to find and fix all incompatibilities.

**Comment by AlirezaRa94** [comment]
> That's because the dynamic config was only done on 20th March ([dbae8f9](https://github.com/e-mission/op-admin-dashboard/commit/dbae8f9992bdee927be54aefcbf2084817f0ae51))

I've implemented dynamic config on this branch (https://github.com/e-mission/op-admin-dashboard/pull/13), and this is just code refactoring.

File: utils/db_utils.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (shankari) [reply utils/db_utils.py:5]: @AlirezaRa94 I took a quick look at the functions - they are for loading uuids and trip tables.

You can use TimeSeries.find_entries() to get a list of entries for a particular time range (TimeQuery), which you can then call `json_normalize` on and then select the columns (= apply the projection) using pandas.

**Comment by shankari** [comment]
> I've implemented dynamic config on this branch (#13), and this is just code refactoring.

@AlirezaRa94 let's not get into a nitpicky discussion about this. The dashboard code was not completely done by end of Feb, but neither was the NREL code for the survey, and you are absolutely right that NREL had a significant delay in reviewing the code after it was written, at least in part because we were scrambling to finish the UI code changes. Hopefully this is not the end of our collaboration, and we will continue to work together on improvements to both the survey and the dashboard.

**Concretely, do you have an ETA for when the data access changes in`db_util` will be done?**

We cannot deploy until they are complete. We can make the related changes ourselves - I wrote the entire server code, I can restructure two data access functions if I need to - but that will take our time away from the app UI changes. The whole point of the feature swap was that UNSW working on the dashboard would free up NREL resources to work on time use...

**Comment by shankari** [comment]
@AlirezaRa94 @swastis10 This is a long and complicated PR already, so I am going to merge this as the first prototype release. It does meet all the minimum requirements:
- login with AWS cognito
- support trip table exports
- support map display

I will then file a series of issues for cleanup as I see them, and for new functionality as we get feedback from program admins. Especially given that @AlirezaRa94 is now working on this project on a limited basis, I think it will be easier to track progress that way.

**Comment by shankari** [comment]
Current issues are:
Switch to the timeseries to read data instead of accessing mongodb directly: https://github.com/e-mission/op-admin-dashboard/issues/29
Infer the config from the URL instead using the `STUDY_NAME/STUDY_CONFIG` environment variable: https://github.com/e-mission/op-admin-dashboard/issues/31
Finalize production config: https://github.com/e-mission/op-admin-dashboard/issues/32

Those seem enough to be starting with; I will add more functionality-based issues as we test more.

**Comment by shankari** [comment]
I have also turned off write access for everybody except me and NREL IT (@asiripanich @AlirezaRa94 @swastis10), all further improvements will be through fork + pull request only. Please let me know if you have concerns about this.
